File: src/main/java/elte/softwaretechnology/stockprices/data/Article.java
package elte.softwaretechnology.stockprices.data;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Article {
	private String snippet;
	private String title;
	private boolean kicker;
	private List<KeyWord> keyWords = new ArrayList<>();
	private LocalDateTime dateOfPublication;
	private Integer wordCount;
	private Integer score; //TODO: check if this should be excluded, or not

	public void addKeyWord(KeyWord keyWord) {
		keyWords.add(keyWord);
	}

	public String getSnippet() {
		return snippet;
	}

	public void setSnippet(String snippet) {
		this.snippet = snippet;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public boolean isKicker() {
		return kicker;
	}

	public void setKicker(boolean kicker) {
		this.kicker = kicker;
	}

	public List<KeyWord> getKeyWords() {
		return keyWords;
	}

	public void setKeyWords(List<KeyWord> keyWords) {
		this.keyWords = keyWords;
	}

	public LocalDateTime getDateOfPublication() {
		return dateOfPublication;
	}

	public void setDateOfPublication(LocalDateTime dateOfPublication) {
		this.dateOfPublication = dateOfPublication;
	}

	public Integer getWordCount() {
		return wordCount;
	}

	public void setWordCount(Integer wordCount) {
		this.wordCount = wordCount;
	}

	public Integer getScore() {
		return score;
	}

	public void setScore(Integer score) {
		this.score = score;
	}
}

File: StockPricePredicter/Article.py
class Article:
    def __init__(self, row):
        if len(row) != 8:
            raise ValueError("Article data should contain 8 fields.")
        self.id = row[0]
        self.date = row[1]
        self.score = row[2]
        self.snippet = row[3]
        self.title = row[4]
        self.url = row[5]
        self.word_count = row[6]
        self.query_result_id = row[7]

    def __eq__(self, other):
        return self.id == other.id

    def __ne__(self, other):
        return not self.__eq__(other)
